src/core/qgsmapsettings.cpp

```diff
@@ -356,6 +356,16 @@ bool QgsMapSettings::testFlag( QgsMapSettings::Flag flag ) const
   return mFlags.testFlag( flag );
 }
 
+void QgsMapSettings::setCustomRenderFlags( const QString &customRenderFlags )
+{
+  mCustomRenderFlags = QVariantMap();
```

Comment:
Only issue with this approach is that existing code which may have been removing an existing flag will no longer function. I think it's safest to leave the deprecated methods working with the pure string mCustomRenderFlags, and keep the deprecated and newer approaches completely isolated from each other...